### PR TITLE
ref(profile): do not overwrite class profile

### PR DESCRIPTION
**Pull request by cmgustavo**

### CODE REVIEW

**Comment by Gamboster** [comment]
ACK

File: src/models/profile/profile.model.ts [hunk 1]
```diff
@@ -12,39 +12,44 @@ export class Profile {
...
-  public create(opts?): Profile {
-    opts = opts ? opts : {};
-    let x = new Profile();
-    x.createdOn = Date.now();
-    x.credentials = opts.credentials || [];
-    x.disclaimerAccepted = false;
-    x.onboardingCompleted = false;
-    x.checked = {};
-    return x;
+  public getObj() {
```
Comment (matiu) [inline src/models/profile/profile.model.ts]: toObject()

File: src/models/profile/profile.model.ts [hunk 2]
```diff
@@ -12,39 +12,44 @@ export class Profile {
...
-    x.credentials = obj.credentials;
-    x.disclaimerAccepted = obj.disclaimerAccepted;
-    x.onboardingCompleted = obj.onboardingCompleted;
-    x.checked = obj.checked || {};
-    x.checkedUA = obj.checkedUA || {};
-
-    if (x.credentials[0] && typeof x.credentials[0] != 'object')
-      throw new Error('credentials should be an object');
-    return x;
+  public create(opts?) {
```
Comment (matiu) [inline src/models/profile/profile.model.ts]: fromObject()

File: src/models/profile/profile.model.ts [hunk 3]
```diff
@@ -12,39 +12,44 @@ export class Profile {
...
+    this.disclaimerAccepted = obj.disclaimerAccepted || false;
+    this.onboardingCompleted = obj.onboardingCompleted || false;
+    this.checked = obj.checked || {};
+    this.checkedUA = obj.checkedUA;
+
+    if (this.credentials[0] && typeof this.credentials[0] != 'object')
+      throw new Error('credentials should be an array of objects');
   }
 
   public toObj(): string {
```
Comment (matiu) [inline src/models/profile/profile.model.ts]: toString()

File: src/providers/profile/profile.ts [hunk 4]
```diff
@@ -1277,7 +1280,7 @@ export class ProfileProvider {
     this.persistenceProvider.removeAllWalletData(walletId);
     this.events.publish('Local/WalletListChange');
 
-    return this.persistenceProvider.storeProfile(this.profile);
+    return this.persistenceProvider.storeProfile(this.profile.getObj());
```
Comment (matiu) [inline src/providers/profile/profile.ts]: this is VERY error-prune.  Passing `this.profile` will also work and will lead to disastrous consequences.

**Comment by matiu** [comment]
ACK

**Pull request merged**